js/components/graph/Graph.js

```diff
@@ -514,14 +515,18 @@ export class Graph extends React.Component {
 
     this.unfocusPrereqs(courseId)
 
-    const timeout = setTimeout(() => {
-      this.setState({ showInfoBox: false })
-    }, 400)
+    setTimeout(() => {
+      if (!this.state.isMouseOnInfoBox) {
+        const timeout = setTimeout(() => {
```

Comment:
There's a subtle bug here: if the user quickly mouses over one course and then a second course (e.g., move the mouse quickly over CSC110 and then CSC111), the info box for CSC111 will disappear. You can learn about aborting timeouts to ensure that this doesn't happen.